src/main/groovy/org/transmartproject/batch/browsetag/BrowseTagTypeDatabaseReader.groovy

```diff
@@ -0,0 +1,180 @@
+package org.transmartproject.batch.browsetag
+
+import groovy.util.logging.Slf4j
+import org.springframework.batch.item.ExecutionContext
+import org.springframework.batch.item.ItemStreamReader
+import org.springframework.batch.item.database.JdbcCursorItemReader
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.jdbc.core.RowMapper
+import org.transmartproject.batch.clinical.db.objects.Tables
+
+import javax.annotation.PostConstruct
+import javax.sql.DataSource
+import java.sql.ResultSet
+import java.sql.SQLException
+
+/**
+ * Gets the browse tag types from database.
+ */
+@Slf4j
+class BrowseTagTypeDatabaseReader implements ItemStreamReader<BrowseTagType> {
+
+    @Delegate
+    JdbcCursorItemReader<BrowseTagType> delegate
+
+    @Autowired
+    DataSource dataSource
+
+    private final Map<String, BrowseFolderType> folderTypes = [:]
+
+    private final Map<Long, BrowseTagType> tagTypes = [:]
```

Comment:
Make it clear that these properties store caches. If they were not classes, you'd have to save them to satisfy the contract of `ItemStreamReader`, which was my first though.

This bean should also be make step scoped, as you otherwise you're wasting memory by having these caches around.